opensearch/src/main/java/org/opensearch/sql/opensearch/data/value/OpenSearchExprValueFactory.java

```diff
@@ -214,7 +214,13 @@ private ExprValue parseStruct(Content content, String prefix) {
    */
   private ExprValue parseArray(Content content, String prefix) {
     List<ExprValue> result = new ArrayList<>();
-    content.array().forEachRemaining(v -> result.add(parse(v, prefix, Optional.of(STRUCT))));
+    content.array().forEachRemaining(v -> {
+      if (v.isString()) {
```

Comment:
Why does this required a special case. Isn't string type handled by parse function.

Reply:
by default it's parsing an array of structs, not an array of strings